webapp/src/ts/modules/analytics/analytics-targets.component.ts

```diff
@@ -109,4 +109,12 @@ export class AnalyticsTargetsComponent implements OnInit, OnDestroy {
         });
       });
   }
+
+  getSubtitleKey(subtitleKey: string, reportingPeriod: ReportingPeriod): string {
+    if (!subtitleKey) {
+      return '';
+    }
+    return reportingPeriod === ReportingPeriod.CURRENT ? 'this_month' : 'all_time';
```

Comment:
This still doesn't cover what we need to do here. The situation now is that you're not even using the `subtitleKey` that is configured. 

What needs to change here is that `subtitleKey` _will be a function_! 
So you'll need to do something like: 

```
return typeof subtitleKey === 'function' ? subtitleKey(reportingPeriod) : subtitleKey;
```